images/tensorflow/Dockerfile

```diff
@@ -20,7 +20,16 @@ ARG image_version
 ARG model_garden_branch="master"
 ARG tensorflow_text_version="nightly"
 
-RUN apt-get update && apt-get install -y --no-install-recommends git
+# TODO (ericlefort) remove this once the base image gets updated to 22.04
+# Keep track of the full set of dependencies in 3.8 and reinstall them on 3.10
+RUN pip3 freeze > requirements.txt
+RUN apt-get install software-properties-common && apt-get update && add-apt-repository ppa:deadsnakes/ppa
```

Comment:
If you're going to reinstall everything, then I strongly suggest starting from the `python:3.10` base image, or an `nvidia/cuda` base image. IMO, the real benefit of using the upstream TF image off of the shelf if that it is tested before we get it on Dockerhub. If you're creating a new python env within that image, I'm not sure how helpful it really is.

Reply:
Yeah, it's definitely not pretty. We do expect the TF team to do this upgrade in the near-term though (still confirming a plan/timeline with them) so it's only a band-aid until then.